typescript/cli/src/utils/chains.ts

```diff
@@ -18,7 +19,7 @@ export async function runSingleChainSelectionStep(
   const chain = (await select({
     message,
     choices,
-    pageSize: 30,
+    pageSize: process.stdout.rows - 2 || DEFAULT_PAGE_SIZE, // subtract 2, to exclude previous prompt text
```

Comment:
There probably should be a minimum. If process.stdout.rows is 0, we would pass in a page size of -2 here

Reply:
Yeah, the intention for `DEFAULT_PAGE_SIZE` was to do that, but it looks like thats not working as intended. Fixed

Reply:
Added calculatePageSize fn